src/codewhisperer/activation.ts

```diff
@@ -244,17 +244,15 @@ export async function activate(context: ExtContext): Promise<void> {
                 context.extensionContext.globalState.get<boolean>(
                     CodeWhispererConstants.accessTokenMigrationDoNotShowAgainKey
                 ) || false
-            const notificationLastShown =
-                context.extensionContext.globalState.get<Date>(
+            const notificationLastShown: number =
+                context.extensionContext.globalState.get<number | undefined>(
                     CodeWhispererConstants.accessTokenMigrationDoNotShowAgainLastShown
-                ) || t
+                ) || 1
 
             //Add 7 days to notificationLastShown to determine whether warn message should show
-            notificationLastShown.setDate(notificationLastShown.getDate() + 7)
-            if (doNotShowAgain || notificationLastShown <= t) {
+            if (doNotShowAgain || notificationLastShown + 1000 * 60 * 60 * 24 * 7 >= Date.now()) {
```

Comment:
unless prettier forced this, I'd use parens to be extra clear about order of operations.
```suggestion
            if (doNotShowAgain || notificationLastShown + (1000 * 60 * 60 * 24 * 7) >= Date.now()) {
```